DotNetHowToAutocomplete/DotNetHowToAutocomplete/Controllers/HomeController.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+
+using Microsoft.Azure.Search;
+using Microsoft.Azure.Search.Models;
+using System.Configuration;
+
+namespace AutocompleteTutorial.Controllers
+{
+    public class HomeController : Controller
+    {
+        private static SearchServiceClient _searchClient;
+        private static ISearchIndexClient _indexClient;
+        private static string IndexName = "nycjobs";
+
+        public static string errorMessage;
+
+        private void InitSearch()
+        {
+                string searchServiceName = ConfigurationManager.AppSettings["SearchServiceName"];
+                string apiKey = ConfigurationManager.AppSettings["SearchServiceApiKey"];
+
+                // Create a reference to the NYCJobs index
+                _searchClient = new SearchServiceClient(searchServiceName, new SearchCredentials(apiKey));
+                _indexClient = _searchClient.Indexes.GetClient(IndexName);
+        }
+
+        public ActionResult Index()
+        {
+            return View();
+        }
+
+        public ActionResult IndexJavaScript()
+        {
+            return View();
+        }
+
+        public ActionResult About()
+        {
+            ViewBag.Message = "Your application description page.";
+
+            return View();
+        }
+
+        public ActionResult Contact()
+        {
+            ViewBag.Message = "Your contact page.";
+
+            return View();
+        }
+        public ActionResult Suggest(bool highlights, bool fuzzy, string term)
+        {
+            InitSearch();
+
+            // Call suggest API and return results
+            SuggestParameters sp = new SuggestParameters()
+            {
+                UseFuzzyMatching = fuzzy,
+                Top = 5
+            };
+
+            if (highlights)
+            {
+                sp.HighlightPreTag = "<b>";
+                sp.HighlightPostTag = "</b>";
+            }
+
+            DocumentSuggestResult resp = _indexClient.Documents.Suggest(term, "sg", sp);
+
+            // Convert the suggest query results to a list that can be displayed in the client.
+            List<string> suggestions = resp.Results.Select(x => x.Text).ToList();
+            return new JsonResult
+            {
+                JsonRequestBehavior = JsonRequestBehavior.AllowGet,
+                Data = suggestions
+            };
+        }
+
+        public ActionResult AutoComplete(string term)
+        {
+            InitSearch();
+            //Call autocomplete API and return results
+            AutocompleteParameters sp = new AutocompleteParameters()
+            {
+                AutocompleteMode = AutocompleteMode.OneTermWithContext,
+                UseFuzzyMatching = false,
+                Top = 5,
+                MinimumCoverage = 80
+            };
+            AutocompleteResult resp = _indexClient.Documents.Autocomplete(term, "sg", sp);
```

Comment:
Same as above; consider making this method `async` so you can call `AutocompleteAsync` here.

Reply:
I'm not sure the Suggest controller method can be async so calling the autocomplete async method seems unnecessary.  Maybe for another sample...